cli/provisioners_test.go

```diff
@@ -0,0 +1,209 @@
+	pd2 := dbgen.ProvisionerDaemon(t, coderdAPI.Database, database.ProvisionerDaemon{
+		Name:       "provisioner-2",
+		CreatedAt:  dbtime.Now().Add(2 * time.Second),
+		LastSeenAt: sql.NullTime{Time: coderdAPI.Clock.Now().Add(-time.Hour), Valid: true},
+		KeyID:      uuid.MustParse(codersdk.ProvisionerKeyIDBuiltIn),
+	})
+	w2 := dbgen.Workspace(t, coderdAPI.Database, database.WorkspaceTable{
+		OwnerID:    memberUser.ID,
+		TemplateID: template.ID,
+	})
+	wb2ID := uuid.MustParse("00000000-0000-0000-dddd-000000000002")
+	job2 := dbgen.ProvisionerJob(t, db, coderdAPI.Pubsub, database.ProvisionerJob{
+		WorkerID:    uuid.NullUUID{UUID: pd2.ID, Valid: true},
+		Input:       json.RawMessage(`{"workspace_build_id":"` + wb2ID.String() + `"}`),
+		CreatedAt:   dbtime.Now().Add(3 * time.Second),
+		StartedAt:   sql.NullTime{Time: coderdAPI.Clock.Now().Add(-2 * time.Hour), Valid: true},
+		CompletedAt: sql.NullTime{Time: coderdAPI.Clock.Now().Add(-time.Hour), Valid: true},
+		Tags:        database.StringMap{"owner": "", "scope": "organization"},
+	})
+	dbgen.WorkspaceBuild(t, coderdAPI.Database, database.WorkspaceBuild{
+		ID:                wb2ID,
+		JobID:             job2.ID,
+		WorkspaceID:       w2.ID,
+		TemplateVersionID: version.ID,
+	})
+
+	// Create a pending job.
+	w3 := dbgen.Workspace(t, coderdAPI.Database, database.WorkspaceTable{
+		OwnerID:    memberUser.ID,
+		TemplateID: template.ID,
+	})
+	wb3ID := uuid.MustParse("00000000-0000-0000-dddd-000000000003")
+	job3 := dbgen.ProvisionerJob(t, db, coderdAPI.Pubsub, database.ProvisionerJob{
+		Input:     json.RawMessage(`{"workspace_build_id":"` + wb3ID.String() + `"}`),
+		CreatedAt: dbtime.Now().Add(4 * time.Second),
+		Tags:      database.StringMap{"owner": "", "scope": "organization"},
+	})
+	dbgen.WorkspaceBuild(t, coderdAPI.Database, database.WorkspaceBuild{
+		ID:                wb3ID,
+		JobID:             job3.ID,
+		WorkspaceID:       w3.ID,
+		TemplateVersionID: version.ID,
+	})
+
+	// Create a provisioner that is idle.
+	_ = dbgen.ProvisionerDaemon(t, coderdAPI.Database, database.ProvisionerDaemon{
+		Name:      "provisioner-3",
+		CreatedAt: dbtime.Now().Add(3 * time.Second),
+		KeyID:     uuid.MustParse(codersdk.ProvisionerKeyIDBuiltIn),
+	})
+
+	updateReplaceUUIDs(coderdAPI)
+
+	for id, replaceID := range replace {
+		t.Logf("replace[%q] = %q", id, replaceID)
+	}
+
+	t.Run("list", func(t *testing.T) {
+		t.Parallel()
+
+		var got bytes.Buffer
+		inv, root := clitest.New(t,
+			"provisioners",
+			"list",
+			"--column", "id,created at,last seen at,name,version,api version,tags,status,current job id,previous job id,previous job status,organization",
+		)
+		inv.Stdout = &got
+		clitest.SetupConfig(t, member, root)
+		err := inv.Run()
+		require.NoError(t, err)
+
+		clitest.TestGoldenFile(t, t.Name(), got.Bytes(), replace)
+	})
+
+	t.Run("jobs list", func(t *testing.T) {
+		t.Parallel()
+
+		var got bytes.Buffer
+		inv, root := clitest.New(t,
+			"provisioners",
```

Comment:
Is it `provisioners` or `provisioner`?

Reply:
The end goal is to show `provisioners` in the help for consistency with our other commands. I think you mentioned to move the rename from provisioner -> provisioners to a follow-up PR in https://github.com/coder/coder/pull/15940 so I implemented it here. Since it's aliased it doesn't really matter what the test calls out, so it'll be one less change in the follow-up PR.